scintillometry/tests/test_folding.py

```diff
@@ -0,0 +1,90 @@
+# Licensed under the GPLv3 - see LICENSE
+"""Full-package tests of pulse folding."""
+
+import pytest
+import numpy as np
+import astropy.units as u
+from astropy.time import Time
+
+from ..base import Task
+from ..generators import EmptyStreamGenerator
+from ..pulse_folding import TimeFold
+from ..functions import Square
+
+
+class TestFoldingBase:
+    def setup(self):
+        self.start_time = Time('2010-11-12T13:14:15')
+        self.sample_rate = 10. * u.kHz
+        self.shape = (6000, 2, 4)
+        self.nh = EmptyStreamGenerator(shape=self.shape,
```

Comment:
Since we don't use the fact that it is empty, how about using `StreamGenerator`?